From 69007ef1222293bd3b03e7d1c23249a39b79b126 Mon Sep 17 00:00:00 2001
From: Michael Abood <person4265@gmail.com>
Date: Mon, 8 Nov 2021 13:01:34 -0500
Subject: [PATCH 193/210] arm64: dts: qcom: samsung-gtelwifiue: Add ZT7554
 touchscreen

Add support for the ZT7554 touchscreen used in samsung-gtelwifiue.

Signed-off-by: Michael Abood <person4265@gmail.com>
---
 .../dts/qcom/apq8016-samsung-gtelwifiue.dts   | 52 +++++++++++++++++++
 1 file changed, 52 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/apq8016-samsung-gtelwifiue.dts b/arch/arm64/boot/dts/qcom/apq8016-samsung-gtelwifiue.dts
index d01c5b8e5c2..fff5ed45340 100644
--- a/arch/arm64/boot/dts/qcom/apq8016-samsung-gtelwifiue.dts
+++ b/arch/arm64/boot/dts/qcom/apq8016-samsung-gtelwifiue.dts
@@ -46,6 +46,21 @@ home {
 			linux,code = <KEY_HOMEPAGE>;
 		};
 	};
+
+	reg_vdd_tsp: regulator-vdd-tsp {
+		compatible = "regulator-fixed";
+		regulator-name = "vdd_tsp";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+
+		gpio = <&msmgpio 73 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&reg_tsp_en_default>;
+
+		startup-delay-us = <150>;
+	};
 };
 
 &blsp_i2c2 {
@@ -72,6 +87,27 @@ accelerometer@1d {
 	};
 };
 
+&blsp_i2c5 {
+	status = "okay";
+
+	touchscreen@20 {
+		compatible = "zinitix,zt7554", "zinitix,bt541";
+		reg = <0x20>;
+		interrupt-parent = <&msmgpio>;
+		interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
+
+		touchscreen-size-x = <800>;
+		touchscreen-size-y = <1280>;
+		linux,keycodes = <KEY_APPSELECT KEY_BACK>;
+
+		vdd-supply = <&reg_vdd_tsp>;
+		vddo-supply = <&pm8916_l16>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&tsp_int_default>;
+	};
+};
+
 &blsp1_uart2 {
 	status = "okay";
 };
@@ -244,4 +280,20 @@ gpio_keys_default: gpio-keys-default {
 		drive-strength = <2>;
 		bias-pull-up;
 	};
+
+	reg_tsp_en_default: reg-tsp-en-default {
+		pins = "gpio73";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	tsp_int_default: tsp-int-default {
+		pins = "gpio13";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
 };
-- 
2.34.1

